### DISCUSSION TITLE
Error getting Copilot to work in visual studio 2022 pro.

### DISCUSSION THREAD

**Discussion by yk-sd** [community/community#50484] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Hello guys,

I am getting issues while trying to activate the github copilot extension in visual studio 2022 pro.

When I press Sign in to activate copilot. I get this error:

[Error] An error occurred for 'RunOperation:RunOperation': StreamJsonRpc.RemoteInvocationException: Request signInInitiate failed with message: self signed certificate in certificate chain
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__145`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Copilot.CopilotClient.<RunOperation>d__20`1.MoveNext()
NotSignedIn

Anyone can help me? I paid for the subscription but am unable to use copilot.

**Comment by ayim** [comment]
Hey, Aaron from Copilot team here.
I haven't run into this issue before, would you be willing to share some crash dumps/hop on a screenshare to go through this together?

Shoot me an email at [username] at github dot com and we'll get the ball rolling.

**Reply by yk-sd** [reply]
Hi,

Sure, I will record a video demostrating the issue and send it to your e-mail.

Thank you

**Comment by yk-sd** [comment, marked as answer]
Fixed.

Solution:

Open Microsoft Edge (or another browser that shows certificate information).
Navigate to the github.com website.
Click on the SSL lock icon next to the address bar.
Click on "Connection is secure".
Click on "Certificate (Valid)".
In the certificate window, select the top-most certificate in the chain (usually the root or intermediate certificate).
Click on "Export" and save the certificate as a .pem file.
Open a Command Prompt as an administrator.
Run the command: (setx NODE_EXTRA_CA_CERTS "C:\path\to\your\certificate.pem"). Replace "C:\path\to\your\certificate.pem" with the actual path to the .pem file that you saved.
Restart Visual Studio.
Copilot should now work.
Note: If you are using a different browser or operating system, the steps may differ slightly, but the general idea is to export the self-signed certificate and set the NODE_EXTRA_CA_CERTS environment variable to point to it.

**Reply by nbelley** [reply]
Hi, it doesn't work for me. Is there a way to list the contect of the NODE_EXTRA_CA_CERTS variable to see if its ok?

**Reply by yk-sd** [reply]
![image](https://user-images.githubusercontent.com/128151346/231967362-7c8cd996-c21e-4c67-87ed-38c192e32d30.png)

https://docs.oracle.com/en/database/oracle/machine-learning/oml4r/1.5.1/oread/creating-and-modifying-environment-variables-on-windows.html#GUID-DD6F9982-60D5-48F6-8270-A27EC53807D0

**Reply by yk-sd** [reply]
Forgot to include documentation :) updated my comment.

**Reply by Alineol** [reply]
open visual studio and click tools>options>GitHub>Copilot> and set "Accept non-trusted certificates automatically" = true
![image](https://user-images.githubusercontent.com/92826153/232141655-f17d3e88-862b-46f8-9e17-286c4eeef78f.png)
after doing it click here to login
![image](https://user-images.githubusercontent.com/92826153/232141991-a0bbfc7b-2708-4565-ba73-33ccaeb0ecd3.png)

**Reply by nbelley** [reply]
@Alineol this doesn't work. I also tried the exta certs thing, I'll try again but for now nothing works.

**Reply by nbelley** [reply]
@yk-sd your solutioon worked! Thanks!

**Reply by tolgabalci** [reply]
> @Alineol this doesn't work. I also tried the exta certs thing, I'll try again but for now nothing works.

Make sure you export the right cert from the cert list.  The default one is definitely NOT the right one.  I clicked on the very top (root) cert and that worked.  Optionally you can have the default one (the page itself) selected, but when you export, under [Save as type] select "certificate chain" and it will add the selected cert and all parent certs to the .pem file.

Also make sure to do the above for github.com and save the cert with .pem extension (You have to manually type that)

(PS: I added the variable as a system variable, but user variable should work as well)

**Comment by NelCapeTown** [comment]
Thank you @yk-sd.  Nice and succinct, and it works!

**Comment by scott-kowalczyk-mckesson** [comment]
We had to add https://*.copilot-proxy.githubusercontent.com/ as a Trusted Site in Internet Options at our company.

**Comment by epolicardo** [comment]
It worked for me also! Thanks @yk-sd !!

**Comment by tomdaniel-mensura** [comment]
Thanks @yk-sd , worked for us as well!

**Comment by borninarage** [comment]
So this has been driving me round the bend for the past month.

I finally got this working assuming something like forcepoint proxy is in your way.

1.  https://*.copilot-proxy.githubusercontent.com/ needs to be in your trusted internet sites. 

![image](https://github.com/community/community/assets/137099949/c017d2c0-9955-4df9-95f3-e8db3e7c726d)

2. In Edge (works for me) need to go to the erroring endpoint in my case https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex 

You need to go to the certificate select the uppermost cert in the chain and export. You need to save as a pem file but make sure you export the whole chain.
a. Click the padlock/cert info, expand the connection secure option and click the little certificate icon
![image](https://github.com/community/community/assets/137099949/6834ac3b-562e-4b03-9b59-8d6084734146)
b. Click to details, in the top window select the upper most certificate and click export
![image](https://github.com/community/community/assets/137099949/e4312d00-1672-4655-b27e-05ca2289b187)
c. Make sure you change the default save as option to certificate chain, and make sure you change the extension from crt to pem.
![image](https://github.com/community/community/assets/137099949/20c40d3b-3b96-4079-b101-6b6c48a7333e)

3. In a command prompt running as admin elevation run: setx NODE_EXTRA_CA_CERTS - PATH TO YOUR CERT
4. 
![image](https://github.com/community/community/assets/137099949/dd20e6ca-4ceb-4de7-8481-6692d256b200)

**Reply by borninarage** [reply]
tada 
![image](https://github.com/community/community/assets/137099949/3c3c7b2b-908e-45d8-b57c-2fbd101672d1)

**Reply by NeilYager** [reply]
This one worked for me (with PyCharm)

**Comment by jacksonsfaria** [comment]
Same problem here... How MS can be so bad at providing a product.
